python/llm/src/ipex_llm/transformers/models/baichuan2.py

```diff
@@ -166,9 +167,8 @@ def baichuan_attention_forward_7b_quantized(
 
     past_key_value = (key_states, value_states) if use_cache else None
 
-    if attention_mask is not None:
-        if attention_mask.dtype == torch.bool:
-            attention_mask.masked_fill_(attention_mask.logical_not(), float("-inf"))
+    invalidInputError(attention_mask is None or attention_mask.dtype != torch.bool,
```

Comment:
I think we can manually turn attention_mask from bool into float ?

Reply:
baichuan2-7b is using float mask, it doesn't use `bool` mask